pulley/src/decode.rs

```diff
@@ -655,3 +682,43 @@ macro_rules! define_extended_decoder {
     };
 }
 for_each_extended_op!(define_extended_decoder);
+
+fn unwrap_uninhabited<T>(res: Result<T, Uninhabited>) -> T {
+    match res {
+        Ok(ok) => ok,
+    }
+}
+
+#[allow(missing_docs)]
+pub mod operands {
+    use super::*;
+
+    macro_rules! define_operands_decoder {
+        (
+            $(
+                $( #[$attr:meta] )*
+                    $snake_name:ident = $name:ident $( {
+                    $(
+                        $( #[$field_attr:meta] )*
+                        $field:ident : $field_ty:ty
+                    ),*
+                } )? ;
+            )*
+        ) => {
+            $(
+                #[allow(unused_variables)]
+                pub fn $snake_name(pc: &mut UnsafeBytecodeStream) -> ($($($field_ty,)*)?) {
+                    ($($(unwrap_uninhabited(<$field_ty>::decode(pc)),)*)?)
+                }
```

Comment:
Can this take a generic `T: BytecodeStream` and return a `Result<(...), T::Error>`? No need to foist the unsafety upon everyone. We can make `unwrap_uninhabited` a method on an exported extension trait, if we want to make it generally available to users of the decoder.

```rust
pub trait UnwrapUninhabited<T> {
    fn unwrap_uninhabited(self) -> T;
}

impl<T> UnwrapUninhabited<T> for Result<T, Uninhabited> {
    ...
}

...

pub fn $snake_name<T>(pc: &mut T) -> Result<($($($field_ty,)*)?), T::Error> {
    ...
}
```